modules/eks-managed-node-group/main.tf

```diff
@@ -238,7 +242,7 @@ resource "aws_launch_template" "this" {
     for_each = toset(["instance", "volume", "network-interface"])
     content {
       resource_type = tag_specifications.key
-      tags          = merge(var.tags, { Name = var.name }, var.launch_template_tags)
+      tags          = merge(local.launch_template_default_tags, { Name = var.name }, var.launch_template_tags)
```

Comment:
why do we need to set this here? from my understanding, only autoscaling groups behave differently when it comes to default tags

Reply:
I double checked what I'd been seeing on Friday, and while the provider default tags do get automatically included in the `launch_template` resource's tags,  they do not get included in the [tag_specifications](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/launch_template#tag-specifications) of the `launch_template`.   Those instead act much like the ASG tag specifications.  Adding the ability to include the default tags to the `launch_template` tag specifications ensures that any instances created from the `launch_template` will also have the default tags set on the instance itself.

Reply:
Would a different variable name make sense to make it clearer?  Maybe `tag_specification_use_default_tags` or something similar?

Reply:
I think for now we leave off the launch template changes - this matches what we have in the autoscaling module (which at some point will get folded into this module) and just add to the autoscaling group

Reply:
I'm very confused.  The module as it exists right now is not able to correctly set tags for resources created by neither an autoscaling group nor a launch template.  Are you saying that only the autoscaling group use case should be fixed?

Reply:
1. We cannot control the autoscaling groups created by the EKS managed node group service so there is no solution for default tags on that resource currently other than using the external autoscaling group tags resource (has to be external to the module https://github.com/terraform-aws-modules/terraform-aws-eks/issues/1558#issuecomment-1030640207)
2. We should mirror what we did in the autoscaling group module for the self-managed node group module https://github.com/terraform-aws-modules/terraform-aws-autoscaling/blob/3c928b8cfb2be46ff756be350fb26c33ee1a640a/main.tf#L10-L15 which is the same as what the Terraform docs specify. We only add the default tags to the autoscaling group resource

Reply:
I understand there are many issues related to default tags right now which is why I am cautious regarding what gets introduced. The Terraform docs only offer the caveat for autoscaling groups https://www.hashicorp.com/blog/default-tags-in-the-terraform-aws-provider and not launch templates and this is what we have mimicked in the autoscaling group module without issue. So for now, I think that is a safe path if we want to introduce that change

Reply:
Ok, unfortunately I need to maintain my company fork with the launch template changes, since that is the issue I needed to solve when I stumbled across the open issue about the asg.   I'll try to get a moment to pull just the asg changes over to a fork on my personal github and submit a new PR.